Lib/fractions.py

```diff
@@ -252,31 +233,52 @@ def __new__(cls, numerator=0, denominator=None):
 
             elif isinstance(numerator, str):
                 # Handle construction from strings.
-                m = _RATIONAL_FORMAT.match(numerator)
-                if m is None:
-                    raise ValueError('Invalid literal for Fraction: %r' %
-                                     numerator)
-                numerator = int(m.group('num') or '0')
-                denom = m.group('denom')
-                if denom:
-                    denominator = int(denom)
-                else:
-                    denominator = 1
-                    decimal = m.group('decimal')
-                    if decimal:
-                        decimal = decimal.replace('_', '')
-                        scale = 10**len(decimal)
-                        numerator = numerator * scale + int(decimal)
-                        denominator *= scale
-                    exp = m.group('exp')
-                    if exp:
-                        exp = int(exp)
-                        if exp >= 0:
-                            numerator *= 10**exp
+                fraction_literal = numerator
+                num, is_fraction_format, denom = fraction_literal.partition('/')
+                try:
+                    num = num.strip()
+                    denom = denom.strip()
+                    if num and denom and denom[0].isdigit():
```

Comment:
```suggestion
                    if num and denom:
```
If `denom[0]` is not a digit, then `int(denom)` will raise a `ValueError`, which is handled. (I did not fully test this though)
Or even:
```suggestion
                    if num:
```

Reply:
No, e.g. "3/+2".